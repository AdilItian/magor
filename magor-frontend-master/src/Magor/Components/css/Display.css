.flex {
    display: flex;
}
.container {
    width: 100%;
    height: 100%;
}
.flexColumn {
    display: flex;
    flex-direction: column;
}
.flexCenterChildren {
    align-items: center;
    justify-content: center;
}
