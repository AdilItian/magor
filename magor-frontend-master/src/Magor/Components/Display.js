import React from 'react'
import './css/Display.css'
import cx from '../Utils/cx'

export const FlexContainer = (props) => (
    <div
        className={cx(...['flex', 'container', props.className])}
        id={props.id}
        style={{ ...props.style }}
    >
        {props.children}
    </div>
)

export const FlexWrap = (props) => (
    <div
        className={cx(['flex', props.className])}
        id={props.id}
        style={{ flexWrap: 'wrap', ...props.style }}
    >
        {props.children}
    </div>
)

export const FlexContainerCentered = (props) => (
    <div
        className={cx([
            'flex',
            !props.stretch && 'container',
            'flexCenterChildren',
            props.className,
        ])}
        id={props.id}
        style={{ ...props.style }}
    >
        {props.children}
    </div>
)

export const VerticalLayout = (props) => (
    <div
        className={cx(['flexColumn', props.className])}
        id={props.id}
        style={{ alignItems: 'center', ...props.style }}
    >
        {props.children}
    </div>
)

export const HorizontalLayout = (props) => (
    <div
        className={cx(['flex', props.className])}
        id={props.id}
        style={props.style}
    >
        {props.children}
    </div>
)
