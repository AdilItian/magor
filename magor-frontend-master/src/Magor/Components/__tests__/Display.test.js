import React from 'react'
import {FlexContainer, FlexContainerCentered, FlexWrap, VerticalLayout, HorizontalLayout} from '../Display'
import renderer from 'react-test-renderer'

test('No undefined inputs', () => {
    expect(FlexContainer).toBeDefined()
    expect(FlexContainerCentered).toBeDefined()
    expect(FlexWrap).toBeDefined()
    expect(VerticalLayout).toBeDefined()
    expect(HorizontalLayout).toBeDefined()
})

test('FlexContainer Snapshot tests', () => {
    const flexContainer = renderer.create(
        <FlexContainer
            id="FlexContainerId"
            className="customClassName"
            style={{background: 'red'}}>
            This is my child
        </FlexContainer>
    )
    expect(flexContainer.toJSON()).toMatchSnapshot()
    expect(flexContainer.toJSON().props.className).toBe('flex container customClassName')
    expect(flexContainer.toJSON().props.id).toBe('FlexContainerId')
    expect(flexContainer.toJSON().props.style).toEqual({background: 'red'})
    expect(flexContainer.toJSON().children[0]).toBe('This is my child')
})

test('FlexContainerCentered Snapshot tests', () => {
    const flexContainerCentered = renderer.create(
        <FlexContainerCentered
            id="FlexContainerId"
            className="customClassName"
            style={{background: 'red'}}>
            This is my child
        </FlexContainerCentered>
    )
    expect(flexContainerCentered.toJSON()).toMatchSnapshot()
    expect(flexContainerCentered.toJSON().props.className).toBe('flex container flexCenterChildren customClassName')
    expect(flexContainerCentered.toJSON().props.id).toBe('FlexContainerId')
    expect(flexContainerCentered.toJSON().props.style).toEqual({background: 'red'})
    expect(flexContainerCentered.toJSON().children[0]).toBe('This is my child')
})

test('FlexWrap Snapshot tests', () => {
    const flexWrap = renderer.create(
        <FlexWrap
            id="FlexWrapId"
            className="customClassName"
            style={{background: 'red'}}>
            This is my child
        </FlexWrap>
    )
    expect(flexWrap.toJSON()).toMatchSnapshot()
    expect(flexWrap.toJSON().props.className).toBe('flex customClassName')
    expect(flexWrap.toJSON().props.id).toBe('FlexWrapId')
    expect(flexWrap.toJSON().props.style).toEqual({flexWrap: 'wrap', background: 'red'})
    expect(flexWrap.toJSON().children[0]).toBe('This is my child')
})

test('VerticalLayout Snapshot tests', () => {
    const verticalLayout = renderer.create(
        <VerticalLayout
            id="VerticalLayoutId"
            className="customClassName"
            style={{background: 'red'}}>
            This is my child
        </VerticalLayout>
    )
    expect(verticalLayout.toJSON()).toMatchSnapshot()
    expect(verticalLayout.toJSON().props.className).toBe('flexColumn customClassName')
    expect(verticalLayout.toJSON().props.id).toBe('VerticalLayoutId')
    expect(verticalLayout.toJSON().props.style).toEqual({alignItems: 'center', background: 'red'})
    expect(verticalLayout.toJSON().children[0]).toBe('This is my child')
})

test('HorizontalLayout Snapshot tests', () => {
    const horizontalLayout = renderer.create(
        <HorizontalLayout
            id="HorizontalLayoutId"
            className="customClassName"
            style={{background: 'red'}}>
            This is my child
        </HorizontalLayout>
    )
    expect(horizontalLayout.toJSON()).toMatchSnapshot()
    expect(horizontalLayout.toJSON().props.className).toBe('flex customClassName')
    expect(horizontalLayout.toJSON().props.id).toBe('HorizontalLayoutId')
    expect(horizontalLayout.toJSON().props.style).toEqual({background: 'red'})
    expect(horizontalLayout.toJSON().children[0]).toBe('This is my child')
})