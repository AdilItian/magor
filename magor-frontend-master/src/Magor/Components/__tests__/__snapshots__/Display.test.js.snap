// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexContainer Snapshot tests 1`] = `
<div
  className="flex container customClassName"
  id="FlexContainerId"
  style={
    Object {
      "background": "red",
    }
  }
>
  This is my child
</div>
`;

exports[`FlexContainerCentered Snapshot tests 1`] = `
<div
  className="flex container flexCenterChildren customClassName"
  id="FlexContainerId"
  style={
    Object {
      "background": "red",
    }
  }
>
  This is my child
</div>
`;

exports[`FlexWrap Snapshot tests 1`] = `
<div
  className="flex customClassName"
  id="FlexWrapId"
  style={
    Object {
      "background": "red",
      "flexWrap": "wrap",
    }
  }
>
  This is my child
</div>
`;

exports[`HorizontalLayout Snapshot tests 1`] = `
<div
  className="flex customClassName"
  id="HorizontalLayoutId"
  style={
    Object {
      "background": "red",
    }
  }
>
  This is my child
</div>
`;

exports[`VerticalLayout Snapshot tests 1`] = `
<div
  className="flexColumn customClassName"
  id="VerticalLayoutId"
  style={
    Object {
      "alignItems": "center",
      "background": "red",
    }
  }
>
  This is my child
</div>
`;
